sqlglot/parser.py

```diff
@@ -3545,6 +3545,12 @@ def _parse_unary(self) -> t.Optional[exp.Expression]:
     def _parse_type(self, parse_interval: bool = True) -> t.Optional[exp.Expression]:
         interval = parse_interval and self._parse_interval()
         if interval:
+            # Convert INTERVAL 'val_1' unit_1 ... 'val_n' unit_n into a sum of intervals
+            while self._match(TokenType.STRING, advance=False):
```

Comment:
can this be a non string?

Reply:
Ah, apparently it can be - good catch!

```
spark-sql (default)> select interval 5 hours 4 minutes;
0 05:04:00.000000000
Time taken: 0.307 seconds, Fetched 1 row(s)
```

Will fix shortly.

Reply:
Addressed in https://github.com/tobymao/sqlglot/pull/2822/commits/36110bed09d80fe11e7f19a0573e8baac2011f18